pkg/tss/nvram.go

```diff
@@ -6,20 +6,41 @@ package tss
 
 import (
 	"crypto/sha1"
+	"fmt"
 	"io"
 
 	tpm1 "github.com/google/go-tpm/tpm"
 	tpm2 "github.com/google/go-tpm/tpm2"
 	tpmutil "github.com/google/go-tpm/tpmutil"
 )
 
-func nvRead12(rwc io.ReadWriteCloser, index, offset, len uint32, ownerPW string) ([]byte, error) {
-	var ownAuth [20]byte
+func nvRead12(rwc io.ReadWriteCloser, index, offset, len uint32, auth string) ([]byte, error) {
+	var ownAuth [20]byte //owner well known
 
-	if ownerPW != "" {
-		ownAuth = sha1.Sum([]byte(ownerPW))
+	// Get TPMInfo
+	indexData, err := tpm1.GetNVIndex(rwc, index)
+	if err != nil {
+		return nil, err
 	}
-	return tpm1.NVReadValue(rwc, index, offset, len, []byte(ownAuth[:20]))
+	if indexData == nil {
+		return nil, fmt.Errorf("index not found")
+	}
+
+	// Check if authData is needed
+	// AuthRead 0x00200000 | OwnerRead 0x00100000
+	needAuthData := 1 >> (indexData.Permission.Attributes ^ (0x00200000 | 0x00100000))
```

Comment:
what are these magic constants? can you name them?

Reply:
oh. sure :)